### ISSUE TITLE
Search broken for Collections

### ISSUE THREAD

**Issue by Propaganistas** [yajra/laravel-datatables#415]
Hi

I'm using Laravel 5.2.21 and laravel-datatables 6.7.1.

When using collections, it seems search is sometimes broken. I'll describe the culprit below.

In `CollectionEngine` line 45, a model's columns are calculated using the attributes on the model by serializing it:
```php
$this->columns = array_keys($this->serialize($collection->first()));
```

This could yield something like this:
```php
array( 'id', 'name', 'title', 'updated_at');
```

However, later on (e.g. line 135-136) the `searchableColumnIndex` method is operating on the Datatable request array, which is custom defined and could look like this (pseudo):
```php
array ('title', 'updated_at');
```

When the column name is fetched by `getColumnNameByIndex` the initial columns array is reused and the index of let's say 'title' is NOT the same as in the datatables array.

Since the indices of both array are not matching, the described methods are mapping the indices in a wrong way and a search on datatables index 0 (title field) will result in searching throughout all IDs.

**Comment by Propaganistas** [comment]
Update: this issue is also present in 6.6.1 but NOT in 6.5.1.

**Comment by yajra** [comment]
Can you please test `dev-develop` branch if it fixes the issue? Must be related to #413.

**Comment by Propaganistas** [comment]
It works properly when using the develop branch.

Good job!

**Comment by yajra** [comment]
Thanks for confirming. Patch released on v6.7.2.
